packer/build.go

```diff
@@ -274,7 +274,25 @@ PostProcessorRunSeqLoop:
 				continue PostProcessorRunSeqLoop
 			}
 
-			keep = keep || corePP.keepInputArtifact
+			keep := defaultKeep
+			// When user has not set keep_input_artifuact
+			// corePP.keepInputArtifact is nil.
+			// In this case, use the keepDefault provided by the postprocessor.
+			// When user _has_ set keep_input_atifact, go with that instead.
+			// Exception: for postprocessors that will fail/become
+			// useless if keep isn't true, heed forceOverride and keep the
+			// input artifact regardless of user preference.
+			if corePP.keepInputArtifact != nil {
+				if defaultKeep && *corePP.keepInputArtifact == false && forceOverride {
```

Comment:
I'm pretty sure the answer is no but, don't we have a case when the artifact *must* be deleted ?
Is it going to be the case in the future too ?

Here we could have :
```suggestion
				if forceOverride && *corePP.keepInputArtifact != defaultKeep {
```

Reply:
I don't foresee ever forcing users to toss out artifacts they may need, and there are some cases where I definitely only want to force the override on Keep. For example, in the Vagrant post-processor, some providers  set forceOverride + keep  and some don't.  The return value is forceOverride: true for all of the providers, but if the default is set to false I want people to still be able to choose to keep the original artifact. 

I intentionally wrote that statement to make sure it would only override when the default keep is true.

Reply:
Gotcha thanks for clarifying 🙂